### PR TITLE
Merge 0.17.0 RC into main

### PR DESCRIPTION
**Pull request by LuqP2**
## Summary

This brings the `0.17.0-rc` release branch back into `main` for final review. The diff is the accumulated RC work rather than a single narrow feature branch.

Highlights:
- Adds the advanced Image Editor workspace, including crop/resize/transform tools, object editing, inspector controls, color picking, text handling refinements, and metadata-preserving save/overwrite flows.
- Adds Compare v2 with multi-image comparison, loupe sync, normalized layer alignment, flicker, delta/heatmap handling, and related interaction tests.
- Improves ComfyUI support with upscale workflow plumbing, MetaHub graph recovery, workspace navigation fixes, and parser/node registry traversal updates.
- Adds media/audio diagnostics and external media handling, plus the MetaHub video metadata parser contract.
- Hardens cache/save behavior for large chunked caches and background image update scheduling after Save As / Overwrite.
- Updates viewer behavior with natural-size zoom, 1:1/Fit modes, zoom state management, and restored grid/library navigation state.

## Review Notes

- `components/ImageModal.tsx` had the PR conflict against latest `main`; it was resolved by keeping the RC viewer zoom behavior (`1:1`, `Fit`, dynamic min/max zoom) while preserving the accessibility tooltip/ARIA improvements from `main`.
- This PR intentionally includes the release branch merge history so the RC context remains reviewable.

## Validation

- Resolved the `components/ImageModal.tsx` merge conflict locally.
- `git diff --check`
- `npx tsc -b`

### CODE REVIEW

**Comment by LuqP2** [comment]
@codex review the latest commits and check whether there are any remaining release-blocking issues beyond the existing unresolved review threads.

**Comment by LuqP2** [comment]
@codex review the current PR for release-blocking issues only. Ignore already fixed review threads and do not comment on minor parser edge cases unless they can cause data loss, crashes, or broken core user flows.

File: __tests__/comfyui-parser.test.ts [hunk 1]
```diff
@@ -216,6 +385,100 @@ describe('ComfyUI Parser - LoRA Workflows', () => {
...
+        class_type: 'JoinStrings',
+        inputs: {
+          delimiter: ', ',
+          string1: ['1', 0],
+          string2: ['2', 0],
+        },
+      },
+    });
+
+    expect(result.prompt).toBe('alpha, beta');
```
[the remark this answers is not in the document]
Comment (LuqP2) [reply __tests__/comfyui-parser.test.ts:480]: @codex this was already addressed in the latest commit. The JoinStrings test fixture now includes a valid terminal `KSampler` path connected through `CLIPTextEncode`, so `resolvePromptFromGraph()` has a sink node to start traversal from.
